src/platform_impl/macos/view.rs

```diff
@@ -433,22 +450,57 @@ extern "C" fn set_marked_text(
             marked_text.initWithString(string);
         };
         *marked_text_ref = marked_text;
+
+        let composed_string = marked_text_ref.clone().string();
+        let slice = slice::from_raw_parts(
+            composed_string.UTF8String() as *const c_uchar,
+            composed_string.len(),
+        );
+        let composed_string = str::from_utf8_unchecked(slice);
+
+        let state_ptr: *mut c_void = *this.get_ivar("winitState");
+        let state = &mut *(state_ptr as *mut ViewState);
+
+        let cursor_position = selected_range.location as i32;
+        (*this).set_ivar("currentCursorPosition", cursor_position);
+
+        // Delete previous marked text, so that we don't see duplicated texts.
+        let is_preediting: bool = *this.get_ivar("isPreediting");
+        if !is_preediting {
+            let previous_cursor_position = *this.get_ivar::<i32>("previousCursorPosition");
+            delete_marked_text(state, previous_cursor_position);
+        }
+
+        for character in composed_string.chars() {
+            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
+                window_id: WindowId(get_window_id(state.ns_window)),
+                event: WindowEvent::ReceivedCharacter(character),
+            }));
+        }
+
+        (*this).set_ivar("previousCursorPosition", cursor_position);
     }
     trace!("Completed `setMarkedText`");
 }
 
 extern "C" fn unmark_text(this: &Object, _sel: Sel) {
     trace!("Triggered `unmarkText`");
     unsafe {
-        let marked_text: id = *this.get_ivar("markedText");
-        let mutable_string = marked_text.mutableString();
-        let _: () = msg_send![mutable_string, setString:""];
+        let _: () = msg_send![this, clearMarkedText];
```

Comment:
If `clearMarkedText` is not a member, you could call it by `clear_marked_text(this)`